src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -5,25 +5,32 @@
 import com.google.common.util.concurrent.MoreExecutors;
 import com.google.protobuf.Struct;
 import io.pinecone.commons.PineconeDataPlaneInterface;
+import io.pinecone.configs.PineconeConnection;
 import io.pinecone.exceptions.PineconeValidationException;
 import io.pinecone.proto.*;
 import io.pinecone.unsigned_indices_model.QueryResponseWithUnsignedIndices;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 
-public class PineconeFutureDataPlaneClient implements PineconeDataPlaneInterface<ListenableFuture<UpsertResponse>,
+public class AsyncIndex implements PineconeDataPlaneInterface<ListenableFuture<UpsertResponse>,
         ListenableFuture<QueryResponseWithUnsignedIndices>, ListenableFuture<FetchResponse>,
         ListenableFuture<UpdateResponse>, ListenableFuture<DeleteResponse>,
-        ListenableFuture<DescribeIndexStatsResponse>> {
+        ListenableFuture<DescribeIndexStatsResponse>>, AutoCloseable {
```

Comment:
Could we have `PineconeDataPlaneInterface extends AutoCloseable` and get the same behavior as if we implemented both here?

I'm not sure if that makes sense given the reason for extension is the `connection` field which is present on the implementing class.

Reply:
We will like to close any opened gRPC connections that can be a part of the object (instance of a class, clients in this case), so it wouldn't be applicable to the interface